app/services/hooks/useUserRequests.ts

```diff
@@ -23,26 +23,39 @@ export const useUserRequests = (month: number, year: number) => {
         )
       }
 
-      // Unnest assignments and swap requests
-      const assignments = assignmentRes.data?.map(({ event, ...rest }) => ({
-        ...rest,
-        event_id: event?.id,
-        start_date: event?.start_time,
-        end_date: event?.end_time,
-      }))
+      // Unnest assignments and add type
+      const assignments =
+        assignmentRes.data?.map(({ event, ...rest }) => ({
+          type: "ASSIGNMENT",
+          ...rest,
+          event_id: event?.id,
+          start_date: event?.start_time,
+          end_date: event?.end_time,
+        })) || []
 
-      const swaps = swapRes.data?.map(({ event, ...rest }) => ({
-        ...rest,
-        event_id: event?.id,
-        start_date: event?.start_time,
-        end_date: event?.end_time,
-      }))
+      // Unnest swaps and add type
+      const swaps =
+        swapRes.data?.map(({ event, ...rest }) => ({
+          type: "SWAP",
+          ...rest,
+          event_id: event?.id,
+          start_date: event?.start_time,
+          end_date: event?.end_time,
+        })) || []
 
-      return {
-        leave: leaveRes.data,
-        assignment: assignments,
-        swap: swaps,
-      }
+      // Add type to leave requests
+      const leave =
+        leaveRes.data?.map((item) => ({
+          type: "LEAVE",
+          ...item,
+        })) || []
+
+      // Combine all arrays into a single flat array
+      return [...assignments, ...swaps, ...leave].sort((a, b) => {
+        const dateA = new Date(a.start_date).getTime()
+        const dateB = new Date(b.start_date).getTime()
+        return dateA - dateB
+      })
```

Comment:
non-blocking but I believe `a.start_date` should already be a Date object, and then you can use `compareAsc` to compare dates (provided by date-fns) instead of casting to an int and doing int comparison

```suggestion
      return [...assignments, ...swaps, ...leave].sort((a, b) =>
        compareAsc(a.start_date, b.start_date),
      )
```